Formatter/Pool.php

```diff
@@ -34,15 +34,13 @@ class Pool
     protected $logger;
 
     /**
-     * @param LoggerInterface|null $logger
      * @param string|null          $defaultFormatter
+     * @param LoggerInterface|null $logger
      */
-    public function __construct(LoggerInterface $logger = null, $defaultFormatter = null)
+    public function __construct($defaultFormatter, LoggerInterface $logger = null)
```

Comment:
> Actually null should be an invalid value in the first place.

That's right. I think I'll have to make another PR on 3.x to fix that, and maybe also add deprecation notices for each of the `TODOs`